.travis/BuildLinux.sh

```diff
@@ -61,18 +49,20 @@ if [ -z "${RUN_CLANG_TIDY}" ] \
         time make test-libs-domain -j2
         time make test-libs-elliptic -j2
         time make test-libs-evolution -j2
-        time make test-libs-numerical-algorithms -j2
     fi
 
     if [[ ${TRAVIS_BUILD_STAGE_NAME} = "Build test libraries" ]]; then
-        time make test-libs-data-structures -j2
+        time make test-libs-numerical-algorithms -j2
+        # Build DataStructures in serial because the DataVector tests
+        # are very memory intensive
+        time make test-libs-data-structures -j1
         time make test-libs-pointwise-functions -j2
-        time make test-libs-other -j2
```

Comment:
Do I correctly understand that with this line being moved to a different Travis build stage, the "Build test libraries" stage no longer builds all test libraries? If so, consider renaming to something more accurate.

Reply:
Yes. @kidder since you came up with the names and these stages, what are your thoughts?

Reply:
"Build most expensive test libraries" is cumbersome but hopefully descriptive?

Reply:
I'm sort of in favor of naming the stages things like `Build stage 1`, `Build stage 2`, `Build stage 3, run tests, clang-tidy, doxygen` but :man_shrugging:

Reply:
yeah since the test libraries are now built in all 3 stages, and we'll likely need to rebalance again in the future, `Build stage N` is fine with me...

Reply:
Okay with me as well